wgpu-core/src/lock/mod.rs

```diff
@@ -0,0 +1,22 @@
+/*! Deadlock prevention via lock ordering.
+
+This module defines wrappers for the lock types used in `wgpu`
+(`Mutex`, `RwLock`, and `SnatchLock`) that use runtime checks to
+ensure that locks are always acquired in a consistent order, ensuring
+that deadlock is impossible. These checks are enabled only in debug
+builds, and when the `"validate-locks"` feature is enabled.
```

Comment:
nitpick: You don't need double-quotes inside the inline code block here. I understand that that's what somebody would literally write in a `features` array element in a `Cargo.toml`, but it's inconsistent with the styling we use at the top level of `wgpu` ([example](https://docs.rs/wgpu/0.19.3/wgpu/index.html)) and `wgpu-core` ([example](https://docs.rs/wgpu-core/0.19.3/wgpu_core/index.html)).

```suggestion
builds, and when the `validate-locks` feature is enabled.
```